README
NAME
    Dist::Zilla::Plugin::Test::PodSpelling - Author tests for POD spelling

VERSION
    version 2.006008

SYNOPSIS
    In "dist.ini":

            [Test::PodSpelling]

    or:

            [Test::PodSpelling]
            directories = docs
            wordlist = Pod::Wordlist
            spell_cmd = aspell list
            stopwords = CPAN
            stopwords = github
            stopwords = stopwords
            stopwords = wordlist

    If you're using "[ExtraTests]" it must come after "[Test::PodSpelling]",
    it's worth noting that this ships in the "[@Basic]" bundle so you may
    have to remove it from that first.

DESCRIPTION
    This is an extension of Dist::Zilla::Plugin::InlineFiles, providing the
    following file:

      xt/author/pod-spell.t - a standard Test::Spelling test

    Test::Spelling will be added as a develop prerequisite.

ATTRIBUTES
  directories
    Additional directories you wish to search for POD spell checking
    purposes. "bin" and "lib" are set by default.

  wordlist
    The module name of a word list you wish to use that works with
    Test::Spelling.

    Defaults to Pod::Wordlist.

  spell_cmd
    If "spell_cmd" is set then "set_spell_cmd( your_spell_command );" is
    added to the test file to allow for custom spell check programs.

    Defaults to nothing.

  stopwords
    If stopwords is set then "add_stopwords( <DATA> )" is added to the test
    file and the words are added after the "__DATA__" section.

    "stopwords" can appear multiple times, one word per line.

    Normally no stopwords are added by default, but author names appearing
    in "dist.ini" are automatically added as stopwords so you don't have to
    add them manually just because they might appear in the "AUTHORS"
    section of the generated POD document. The same goes for contributors
    listed under the 'x_contributors' field on your distributions META file.

METHODS
  add_stopword
    Called to add stopwords to the stopwords array. It is used to determine
    if automagically detected words are valid and print out debug logging
    for the process.

BUGS
    Please report any bugs or feature requests on the bugtracker website
    https://github.com/xenoterracide/dist-zilla-plugin-test-podspelling/issu
    es

    When submitting a bug or request, please include a test-file or a patch
    to an existing test-file that illustrates the bug or desired feature.

CONTRIBUTORS
    *   Alexandr Ciornii <alexchorny@gmail.com>

    *   Breno G. de Oliveira <garu@cpan.org>

    *   David Golden <dagolden@cpan.org>

    *   Graham Knop <haarg@haarg.org>

    *   Harley Pig <harleypig@gmail.com>

    *   Karen Etheridge <ether@cpan.org>

    *   Randy Stauner <rwstauner@cpan.org>

AUTHORS
    *   Caleb Cushing <xenoterracide@gmail.com>

    *   Marcel Gruenauer <hanekomu@gmail.com>

COPYRIGHT AND LICENSE
    This software is Copyright (c) 2014 by Caleb Cushing.

    This is free software, licensed under:

      The Artistic License 2.0 (GPL Compatible)



Changes
Revision history for Perl extension Dist-Zilla-Plugin-Test-PodSpelling

2.006008  2014-06-17
	- register Test::Spelling as a development dependency ( haarg )

2.006007  2014-03-05
	- munge the file properly even if [ExtraTests] has already moved it

2.006006  2014-02-27
	- require Dist::Zilla 5
	  Unicode Changes + 2.6.3 changes means tests only work on dzil 5+

2.006005  2014-02-23
	- Document ExtraTests problem

2.006004  2014-02-23
	- remove dependency on Path::Class in favor of Path::Tiny

2.006003  2014-02-23
        - prevent distmeta from being populated too early by postponing
          calling it until the file munging phase

2.006002  2013-12-19
	- fix problem with compile tests stalling on windows (chorny) GH #20

2.006001  2013-09-07
	- stop depending on hanekomu word list. Pod::Wordlist is maintained,
	  please request words to be added there
	- remove hack to prevent dzil from adding generated test deps to your dist
	  that consumes this.

2.006000  2013-05-01
	- add contributers to the stopwords list

2.004004  2013-04-29
	- fix tests broken by Test::Spelling 0.18 which changed Test::Builder
	  level

2.004003  2013-03-04
	- fix ::PodSpellingTests version metadata being wrong

2.004002  2013-03-01
	- checked.t should now work on windows if Test::Spelling 0.17+ installed

2.004001  2013-02-14
	- Add documentation for directories feature

2.004000  2013-02-14
	- Add support for customizable search directories GH #8
	- fix bug where unicode names weren't being added properly as stopwords
	  GH #9

2.002006  2012-12-29
	- FIXME: skip checked.t for windows, it hangs see Github issue #7

2.002005  2012-06-27
	- fix test for perls < 5.10 where qr//m did not work properly

2.002004  2012-04-04
	- appears that I forgot to merge 2.002002 before releasing 2.002003

2.002003  2012-04-04
	- make tests required for authors but no one else

2.002002  2012-03-07
	- fix issue with a dash in the copyrigh holder GH #5
	- use Test::Requires over eval

2.002001  2012-01-10
	- fix copyright year

2.002000  2012-01-10
	- Generate stopwords based on the file names in the distribution

2.001004  2011-12-13
	- Fix tests so that spell check passes ( rwstauner )
	- Test test contents ( rwstauner )

2.001003  2011-12-10
	- skip checked if OS is BSD, no idea why the tests fail predominantly on
	  BSD

2.001002  2011-11-27
	- remove issue where duplicate stopwords could be created
	- attempt to fix issue with spellchecking engines not working

2.001001  2011-10-26
	- skip test if no spellchecking engine found ( rwstauner )

2.1.0     2011-10-08
	[FEATURES]
	- now checks your bin/ scripts for pod spelling errors
	[OTHER]
	- added tests

v2.0.0    2011-08-06
	[FEATURES]
	- automatically add copyright owner to stopwords
	- now an author test
	- use semantic versioning
	[DEPRECATED]
	- use of PodSpellingTests Plugin, use Test::PodSpelling instead.
	[OTHER]
	- ownership passes to xenoterracide
	- Dist Name changed with Module name

1.111520  2011-06-01
	- Require Test::Spelling 0.12 in the injected test (thanks doherty for the
	  suggestion)

1.111480  2011-05-28
	- automatically add author names from dist.ini to stopwords (thanks
	  xenoterracide for the suggestion)

1.103491  2010-12-15

1.103490  2010-12-15
	- added configuration (wordlist, spell_cmd, stopwords) to this plugin
	  (thanks harleypig)

1.101420  2010-05-22
	- removed weaver.ini since that's handled in Dist::Zilla's [@MARCEL] now
	- list Test::Spelling in dist.ini so we don't use() it anymore

1.100690  2010-03-10
	- original version


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.019.
use strict;
use warnings;

use 5.010;

use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Author tests for POD spelling",
  "AUTHOR" => "Caleb Cushing <xenoterracide\@gmail.com>, Marcel Gruenauer <hanekomu\@gmail.com>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-Test-PodSpelling",
  "EXE_FILES" => [],
  "LICENSE" => "artistic_2",
  "NAME" => "Dist::Zilla::Plugin::Test::PodSpelling",
  "PREREQ_PM" => {
    "Dist::Zilla" => 5,
    "Dist::Zilla::Plugin::InlineFiles" => 0,
    "Dist::Zilla::Role::FileFinderUser" => 0,
    "Dist::Zilla::Role::FileMunger" => 0,
    "Dist::Zilla::Role::PrereqSource" => 0,
    "Dist::Zilla::Role::TextTemplate" => 0,
    "Moose" => 0,
    "Test::Spelling" => "0.12",
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Cwd" => 0,
    "Dist::Zilla::Tester" => 0,
    "English" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "File::Spec::Functions" => 0,
    "File::pushd" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "List::Util" => 0,
    "Moose::Util" => 0,
    "Path::Tiny" => 0,
    "Test::DZil" => 0,
    "Test::Fatal" => 0,
    "Test::More" => "0.88",
    "Test::Requires" => 0,
    "Test::Script" => "1.05",
    "Test::Tester" => 0,
    "utf8" => 0,
    "version" => 0
  },
  "VERSION" => "2.006008",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Cwd" => 0,
  "Dist::Zilla" => 5,
  "Dist::Zilla::Plugin::InlineFiles" => 0,
  "Dist::Zilla::Role::FileFinderUser" => 0,
  "Dist::Zilla::Role::FileMunger" => 0,
  "Dist::Zilla::Role::PrereqSource" => 0,
  "Dist::Zilla::Role::TextTemplate" => 0,
  "Dist::Zilla::Tester" => 0,
  "English" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "File::Spec::Functions" => 0,
  "File::pushd" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "List::Util" => 0,
  "Moose" => 0,
  "Moose::Util" => 0,
  "Path::Tiny" => 0,
  "Test::DZil" => 0,
  "Test::Fatal" => 0,
  "Test::More" => "0.88",
  "Test::Requires" => 0,
  "Test::Script" => "1.05",
  "Test::Spelling" => "0.12",
  "Test::Tester" => 0,
  "strict" => 0,
  "utf8" => 0,
  "version" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



